feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-ui-key-generator.component.ts

```diff
@@ -0,0 +1,154 @@
+import { Configurator } from '../../../../core/model/configurator.model';
+
+/**
+ * Service to provide unique keys for elements on the UI and for sending to configurator
+ */
+export class ConfiguratorUiKeyGeneratorComponent {
```

Comment:
Although it's named 'component', this is rather a service and should be injected